src/experiments/74.particles-emitter.tsx

```diff
@@ -0,0 +1,192 @@
+import { useControls } from 'leva'
+import React, { useEffect, useMemo, useRef } from 'react'
+
+import { HTMLLayout } from '~/components/layout/html-layout'
+
+function hexToRgb(hex: string) {
+  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex)
+  if (!result || !result[1] || !result[2] || !result[3]) return null
+  return `${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
+    result[3],
+    16
+  )}`
+}
+
+interface Particle {
+  x: number
+  y: number
+  size: number
+  speedX: number
+  speedY: number
+  color: string
+  opacity: number
+}
+
+function ParticlesEmitter() {
+  const containerRef = useRef<HTMLDivElement>(null)
+  const mousePosition = useRef({ x: 0, y: 0 })
+  const particles = useRef<Particle[]>([])
+  const animationRef = useRef<number>()
+
+  /* Controls */
+  const {
+    particleCreationRate,
+    particleSize,
+    particleColor,
+    gravityStrength,
+    particleLifespan,
+    emissionRadius,
+    maxParticles,
+    updateFrequency
+  } = useControls({
+    particleCreationRate: { value: 1, min: 0.1, max: 1, step: 0.1 },
+    particleSize: { value: 4, min: 1, max: 10, step: 1 },
+    particleColor: '#FF4D00',
+    gravityStrength: { value: 0.2, min: -1, max: 1, step: 0.01 },
+    particleLifespan: { value: 40, min: 10, max: 100, step: 10 },
+    emissionRadius: { value: 1, min: 1, max: 10, step: 0.1 },
+    maxParticles: { value: 60, min: 10, max: 200, step: 10 },
+    updateFrequency: { value: 1, min: 1, max: 2, step: 1 }
+  })
+
+  /* Mouse Position */
+  useEffect(() => {
+    const updateMousePosition = (e: MouseEvent) => {
+      mousePosition.current = { x: e.clientX, y: e.clientY }
+    }
+
+    const updateTouchPosition = (e: TouchEvent) => {
+      mousePosition.current = {
+        x: e.touches[0].clientX,
+        y: e.touches[0].clientY
+      }
+    }
+
+    window.addEventListener('mousemove', updateMousePosition)
+    window.addEventListener('touchmove', updateTouchPosition)
+
+    return () => {
+      window.removeEventListener('mousemove', updateMousePosition)
+      window.removeEventListener('touchmove', updateTouchPosition)
+    }
+  }, [])
+
+  const color = useMemo(
+    () => hexToRgb(particleColor) || '187, 134, 255',
+    [particleColor]
+  )
+
+  /* Particles */
+  useEffect(() => {
+    const container = containerRef.current
+    if (!container) return
+
+    let frameCount = 0
+
+    const renderParticles = () => {
+      frameCount++
+
+      if (frameCount % updateFrequency === 0) {
+        let shadow = ''
+
+        particles.current = particles.current.filter((p, i) => {
+          // gravity
+          if (gravityStrength > 0) p.speedY += gravityStrength * 0.98
```

Comment:
let's add a different gravity value to control the gravity toward the floor

they could be called `gravity` and `cursorGravity`